### PR TITLE
'Add models' docs

### PR DESCRIPTION
**Pull request by rasbt**
Adds a checklist guide for those who want to contribute new models to LitGPT.

### CODE REVIEW

File: tutorials/developer-docs/adding-models.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+     mlp_class_name="LLaMAMLP",
+     intermediate_size=28672,
+     rope_base=500000,
+ ),
+```
+
+&nbsp;
+
+> [!NOTE]
+> Some models may require you to implement a new MLP class analogous to `class LLaMAMLP`.
```
Comment (Andrei-Aksionov) [inline tutorials/developer-docs/adding-models.md:89]: A good indicator is existing modeling.py file.
If it exists (in the model's repo), than something in the current version of HF doesn't support the new model and that will require additional changes besides just updating the config.

File: tutorials/developer-docs/adding-models.md [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+&nbsp;
+## 8. Add unit tests
+
+&nbsp;
+### 8.1 Add model unit tests
+
+Open the [`../tests/test_model.py`](../tests/test_model.py) file and add a new `def test_against_hf_...` function using one of the existing functions as a template. For instance,
+
+```
```
Comment (Andrei-Aksionov) [inline tutorials/developer-docs/adding-models.md:190]:
```suggestion
```python
```

File: tutorials/developer-docs/adding-models.md (hunk 1 above)
Comment (rasbt) [reply tutorials/developer-docs/adding-models.md:89]: Good idea, I added this!

File: tutorials/developer-docs/adding-models.md [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+     rope_base=500000,
+ ),
+```
+
+&nbsp;
+
+> [!NOTE]
+> Some models may require you to implement a new MLP class analogous to `class LLaMAMLP`.
+> A reliable indicator is the presence of a `modeling.py` file in the model's original repository.
+> If this file exists, it suggests that the current version of Hugging Face does not support
```
Comment (Andrei-Aksionov) [inline tutorials/developer-docs/adding-models.md:91]: The problem isn't that HF doesn't support it, but that the file contains logic that wasn't met before in any other architecture.

Comment (rasbt) [reply tutorials/developer-docs/adding-models.md:91]: Yes, that's fair, the wording was a bit misleading.

**Pull request merged**